bot/cogs/bot.py

```diff
@@ -81,6 +81,24 @@ def __init__(self, bot: AutoShardedBot):
 
         await ctx.invoke(self.info)
 
+    def repl_stripping(self, msg: str):
+        """
+        Strip msg in order to extract Python code out of REPL output.
+
+        Tries to strip out REPL Python code out of msg and returns the stripped msg.
+        """
+        final = ""
+        for line in msg.splitlines(keepends=True):
+            if line.startswith(">>>") or line.startswith("..."):
+                final += line[4:]
+        log.trace(f"Formatted: {msg} to {final}")
+        if not final:
+            log.debug(f"Found no REPL code in {msg}")
+            return msg
+        else:
+            log.debug(f"Found REPL code in {msg}")
+            return final
```

Comment:
if you did `return final.strip()` here, you'd prevent the following situation:

![image](https://user-images.githubusercontent.com/2098517/39328442-b4f3969c-499b-11e8-8726-f0d1193f3c8f.png)

note the blank line at the end in the example given by the bot.

Reply:
if the code starts with a bunch of spaces, e.g. someone pasting indented code, won't this result in the indentation on the first line being ruined? `.rstrip()` might be preferable?

Reply:
good point. you'll need to do rstrip, nix.

Reply:
Ok